src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail = new Node<>(tail, value, null);
```

Comment:
```suggestion
        tail = new Node<>(tail, value, null);
        if (isEmpty()) {
            tail = head;
        } else {
```
по ідеї це можна винести поза if else так як якщо, ліст пусти то tail = null, тому         tail = new Node<>(tail, value, null); теж саме що і         tail = new Node<>(null, value, null);